### DISCUSSION TITLE
dotnet-sdk dependency -- detect "rollForward" settings?

### DISCUSSION THREAD

**Discussion by SeanKilleen** [renovatebot/renovate#27296] [Suggest an Idea]
### Tell us more.

Hi all,

Love RenovateBot!

Something I noticed recently -- 

When `dotnet-sdk` dependency updates are detected, RenovateBot attempts to update our global.json. 
However, we use `"rollForward": "latestFeature"` in our global.json. This is because otherwise, developers would have to update their local .NET SDK in order to develop on the solution, which most of the time isn't necessary.

Is there a way to get RenovateBot to respect the existing global.json logic? I could ignore the check, but I think I'd still appreciate getting updates as long as they respect our global.json preferences.

Thanks for considering!

**Comment by viceice** [comment]
you can simply disable patch updates as workaround. remember to also enable separate minor and patch.

https://learn.microsoft.com/en-us/dotnet/core/tools/global-json

you should provide samples for what renovate should do for specific roll forward values.

**Reply by SeanKilleen** [reply]
I like the idea of disabling patch updates; that makes sense. Newer to RenovateBot (came over from Dependabot a week ago) so still getting used to that.

My example scenario is essentially: "if the rollForward value would encompass the new version, don't propose the update". In my specific case above, that would mean no updates would be proposed until 9.x is released, because any 8.x versions would be covered by the rollForward in the global.json.

**Reply by rarkins** [reply]
Sounds like rangeStrategy=replace equivalent

**Reply by viceice** [reply]
I think we need a new versioning for this, which implicit converts the version to a range corresponding to the roll forward value.
it would be `^8.0.100` in this case

**Reply by rarkins** [reply]
I created this feature request: https://github.com/renovatebot/renovate/issues/27312
